Frontend/src/app/dataset-list/dataset-list.component.spec.ts

```diff
@@ -1,17 +1,60 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
-
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
 import { DatasetListComponent } from './dataset-list.component';
+import { ActivatedRoute, convertToParamMap, Router } from '@angular/router';
+import { of, throwError } from 'rxjs';
+import { DatasetService } from '../dataset.service';
+import { MatDialogModule, MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { MatTableModule } from '@angular/material/table';
+import { MatSortModule } from '@angular/material/sort';
+import { ReactiveFormsModule } from '@angular/forms';
+import { ProgressSpinnerComponent } from '../progress-spinner/progress-spinner.component';
+import { MatToolbarModule } from '@angular/material/toolbar';
+import { MatIconModule } from '@angular/material/icon';
+import { BrowserAnimationsModule, NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { PopUpWindowComponent } from '../pop-up-window/pop-up-window.component';
+import { HttpClientModule } from '@angular/common/http';
 
 describe('DatasetListComponent', () => {
   let component: DatasetListComponent;
   let fixture: ComponentFixture<DatasetListComponent>;
+  let mockDatasetService: jasmine.SpyObj<DatasetService>;
+  let mockRouter: Router;
+
+
+
+  beforeEach(waitForAsync(() => {
+    mockDatasetService = jasmine.createSpyObj('DatasetService', ['getDatasetForResource']);
+    mockRouter = jasmine.createSpyObj('Router', ['navigate']);
```

Comment:
Nem használod ezek után sehol, el lehet távolítani a mockRoutert innen és a 21-es sorból is

Reply:
Javítva